src/core/mapping/base_mapper.cc

```diff
@@ -134,23 +134,33 @@ void BaseMapper::select_task_options(const Legion::Mapping::MapperContext ctx,
                                      const Legion::Task& task,
                                      TaskOptions& output)
 {
+  Task legate_task(&task, context, runtime, ctx);
 #ifdef LEGATE_USE_COLLECTIVE
-  for (uint32_t idx = 0; idx < task.regions.size(); ++idx) {
-    auto& req = task.regions[idx];
+  auto hi = task.index_domain.hi();
+  auto lo = task.index_domain.lo();
+  for (auto& store : legate_task.inputs()) {
+    if (store.is_future()) continue;
+    auto idx                           = store.requirement_index();
+    std::vector<int32_t> promoted_dims = store.find_imaginary_dims();
+    for (auto& d : promoted_dims) {
+      if ((hi[d] - lo[d]) >= 1) {
+        output.check_collective_regions.insert(idx);
```

Comment:
```suggestion
    std::vector<int32_t> promoted_dims = store.find_imaginary_dims();
    for (auto& d : promoted_dims) {
      if ((hi[d] - lo[d]) >= 1) {
        output.check_collective_regions.insert(store.requirement_index());
```

Reply:
this is definitely a nitpick. in general I believe it's a good writing practice to put the definition close to the use, unless that leads to a lot of repetitions.